src/components/DataSourceSelector.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <section>
+    <div v-if="dataSources.length && !selectedDataSource">
+      <Select2 :dataSources="dataSources" :selectedDataSource="selectedDataSource" @selectDataSource="setDataSource"></Select2>
+      <span @click="showAllDataSources" class="btn-link create-dataSource">Create new data source</span>
+      <div class="checkbox checkbox-icon">
+        <input @change="showAllDataSources" :checked="showAll" type="checkbox" name="showAll" id="showAll" />
+        <label for="showAll">
+          <span class="check">
+            <i class="fa fa-check"></i>
+          </span>
+          Show all data sources
+        </label>
+      </div>
+    </div>
+    <div v-else-if="dataSources.length && selectedDataSource && !changeDataSource">
+      <p>{{ selectedDataSource.id }}. {{ selectedDataSource.name }} <span @click="() => { this.$emit('onDataSourceChange') }" class="btn-link change-dataSource">Change</span></p>
+      <div @click="viewDataSource" class="btn btn-default view-ds-btn">View data source</div>
+    </div>
+    <div v-else-if="dataSources.length && selectedDataSource && changeDataSource">
+      <Select2 :dataSources="dataSources" :selectedDataSource="selectedDataSource" @selectDataSource="setDataSource"></Select2>
+      <span @click="() => { this.$emit('onDataSourceCreate') }" class="btn-link create-dataSource">Create new data source</span>
+      <div class="checkbox checkbox-icon">
+        <input @change="showAllDataSources" :checked="showAll" type="checkbox" name="showAll" id="showAll" />
+        <label for="showAll">
+          <span class="check">
+            <i class="fa fa-check"></i>
+          </span>
+          Show all data sources
+        </label>
+      </div>
+      <div @click="viewDataSource" class="btn btn-default view-ds-btn">View data source</div>
+    </div>
+  </section>
+</template>
+
+<script>
+import Select2 from './Select2';
+
+export default {
+  data() {
+    return {
+      dataSources: []
+    };
+  },
+  props: {
+    currentAppDataSources: {
+      type: Array,
+      default() {
+        return [];
+      }
+    },
+    otherDataSources: {
+      type: Array,
+      default() {
+        return [];
+      }
+    },
+    selectedDataSource: {
+      type: Object,
+      default() {
+        return {};
+      }
+    },
+    changeDataSource: {
+      type: Boolean,
+      default: false
+    },
+    showAll: {
+      type: Boolean,
+      default: false
+    }
+  },
+  methods: {
+    showAllDataSources: function() {
+      this.$emit('onShowAll', !this.showAll);
+      this.prepareData();
+    },
+    prepareData: function() {
+      // If otherDataSources array is empty it means that we show user only ds's for current app
```

Comment:
Avoid abbreviations with **data source(s)**.

```suggestion
      // If otherDataSources array is empty it means that we show user only data sources for current app
```